test/decl/func/functions.swift

```diff
@@ -179,3 +179,12 @@ func parentheticalInout2(_ fn: (((inout Int)), Int) -> ()) {
   var value = 0
   fn(&value, 0)
 }
+
+//===--- Function declaration with uninhabited parameter type
+                                   
+func foo(baz: Never) -> Int { // expected-warning{{will never be executed}}
```

Comment:
Hm, I wish we could get this to appear on the *next* line, but that doesn't mean we can't take this PR.

Reply:
I think I may have to fix it as the tests failed because I didn't put the warning on the return statement?

Reply:
Fixed.